cupyx/fallback_mode/ndarray.py

```diff
@@ -0,0 +1,130 @@
+"""
+class ndarray is wrapper around cupy.ndarray
+to support fallback of methods of type `ndarray.func()`
+"""
+import numpy as np
+
+import cupy as cp
+from cupyx.fallback_mode import data_transfer
+
+try:
+    from cupyx.fallback_mode import fallback
+except ImportError:
+    import sys
+    fallback = sys.modules[__package__ + '.fallback']
+
+
+class ndarray:
+    """
+    Wrapper around cupy.ndarray
+    Gets initialized with a cupy ndarray.
+    """
+
+    def __init__(self, array):
+        self._array = array
+
+    def __getattr__(self, attr):
+        """
+        Catches attributes corresponding to ndarray.
+
+        Args:
+            attr (str): Attribute of ndarray class.
+
+        Returns:
+            (_RecursiveAttr object, self._array.attr):
+            Returns_RecursiveAttr object with numpy_object, cupy_object.
+            Returns self._array.attr if attr is not callable.
+        """
+
+        cupy_object = getattr(cp.ndarray, attr, None)
+
+        numpy_object = getattr(np.ndarray, attr)
+
+        if not callable(numpy_object):
+            return getattr(self._array, attr)
+
+        return fallback._RecursiveAttr(numpy_object, cupy_object, self)
+
+    def _get_array(self):
+        """
+        Returns _array (cupy.ndarray) of ndarray object.
+        """
+        return self._array
+
+    @property
+    def __class__(self):
+        """
+        Redirects __class__ of ndarray object to cupy.ndarray
+        Used while execution of isinstance() method.
+        """
+        return cp.ndarray
+
+
+def _get_cupy_ndarray(args, kwargs):
```

Comment:
This function name is not inappropriate because this function sometimes does not return `ndarray`.

Reply:
`_get_cupy_args` and similarly `_get_fallback_args`?

Reply:
This function is called from only here: https://github.com/cupy/cupy/blob/3e8433d2f5400cb53c92b36ce684506062c4c277/cupyx/fallback_mode/ndarray.py#L68
Do we need to make this function? I think it's OK to write in `make_method` directly.

Reply:
Am.. It's also being used in `_call_numpy`

Reply:
OK. Let's rename to `_get_cupy_args` and move this function to `data_transfer.py`

Reply:
I thought moving to `data_transfer.py` but, `_get_cupy_args` needs `fallback_mode.ndarray` as dependency. This will result in combining `data_transfer.py` along with other three files(as discussed [here](https://github.com/cupy/cupy/pull/2272#discussion_r300683108)).

Reply:
It seems better to call `data_transfer._get_xp_args(ndarray, ndarray._get_array, (args, kwargs))` directly without defining `_get_cupy_args`.